app/views/products/_form.html.erb

```diff
@@ -0,0 +1,68 @@
+<section>
+  <%= form_with model: @product do |f| %>
+    <div class=" form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :name %>
+        <%= f.text_field :name, class: "form-control" %>
+      </div>
+    </div>
+
+    <div class=" form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :stock %>
+        <%= f.text_field :stock, class: "form-control" %>
+      </div>
+    </div>
+
+    <div class=" form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :price %>
+        <%= f.text_field :price, class: "form-control" %>
+      </div>
+    </div>
+
+    <div class=" form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :photo_url %>
+        <%= f.text_field :photo_url, class: "form-control" %>
+      </div>
+    </div>
+
+    <div class=" form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :description %>
+        <%= f.text_area :description, class: "form-control" %>
+      </div>
+    </div>
+
+
+    <div class=" mb-3">
+      <%= f.label :categories %>
+      <%= collection_check_boxes(:product, :category_ids, Category.all, :id, :name) do |b|%>
+        <%= b.label(class:"label-checkbox") {b.check_box + b.text }%>
+      <% end %>
+    </div>
+
+
+    <div class="form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.submit action_name, class: "btn btn-primary" %>
+      </div>
+    </div>
+  <% end %>
+
+  <%= form_with model: @category, url: categories_path, method: :post do |f| %>
+    <div class="form-row">
+      <div class="col-md-4 mb-3">
+        <%= f.label :new_category %>
+        <%= f.text_field :name, class: "form-control" %>
+      </div>
+    </div>
+
+    <div class="form-row">
```

Comment:
Marj requested feedback: The styles are beautiful! One thing I would consider is moving the make new category functionality into the same form as the make new product. The user experience is a little funny since if you enter all the information for a new product and then want to add a new category, when you click the make new category button, your product information is cleared.